api/services/course.js

```diff
@@ -86,8 +88,62 @@ const getCoursesForCarousel = async () => {
   });
 };
 
+const postSignatureUserCourse = async (courseId, authorization) => {
+  if (
+    typeof parseInt(courseId) != 'number' ||
+    parseInt(courseId) !== parseInt(courseId)
+  ) {
+    throw new errors.SignatureUserCourseError('Invalid parament courseId');
+  }
+
+  const { id: userId } = jwtDecode(authorization);
+  const [signature, created] = await db.usersCourses.findOrCreate({
+    where: {
+      userId,
+      courseId,
+    },
+  });
+
+  if (!created)
+    throw new errors.SignatureUserCourseError('User is signatured already');
+
+  return BaseResponse.responseBuilder({
+    status: 200,
+    message: 'User is signatured',
+    data: signature,
+  });
+};
+
+const getSignatureUserCourse = async (courseId, authorization) => {
+  if (
+    typeof parseInt(courseId) != 'number' ||
+    parseInt(courseId) !== parseInt(courseId)
+  ) {
+    throw new errors.SignatureUserCourseError('Invalid parament courseId');
+  }
+
+  const { id: userId } = jwtDecode(authorization);
+
+  const signature = await db.usersCourses.findOne({
+    where: {
+      userId,
+      courseId,
+    },
+  });
+
+  const response = {
+    status: 200,
+    message: (!signature) ? 'User is not signatured' : 'User have been signatured already',
```

Comment:
User is not signatured -> User didn't subscribe / User haven't subscribed 
User have been signatured already -> User has already signed

Reply:
Done